### ISSUE TITLE
Prisma Error reported from a query but the error is from a different query

### ISSUE THREAD

**Issue by thesunny** [prisma/prisma#2753]
## Bug description

I get the an error from a query but the error itself is not the issue. The purpose of this bug submission is that the error is from a different Prisma query than the one that is reporting the error. The error is `received null value for field last_sent_user which may not be null`; however, this field is not present in the call to `prisma.users.findOne` as shown below.

The error is from another method called in parallel using `Promise.all`.

```
Error at path "api/shared/me.ts" on method "entry": 
Invalid `prisma.users.findOne()` invocation in
/Users/sunnyhirai/Projects/airshelf/api/shared/me.ts:5:35

  1 import { API } from "~/api-server"
  2 
  3 const entry = API.method({}, async ([object Object],[object Object],[object Object], ,[object Object], prisma,[object Object], by,[object Object], at ,[object Object]) => {
  4   if (by == null) return {}
→ 5   const user = await prisma.users.findOne(
  PANIC: Application logic invariant error: received null value for field last_sent_user which may not be null

This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.
```

## How to reproduce

I don't have a minimal reproduction as I'm not sure yet what the actual error is (everything appears correct in the call but I shall submit a separate bug report); however, it does happen within the same `Promise.all`.

## Expected behavior

Two potential ways to improve this.

1. The best fix is that `prisma.users.findOne` should not crash when an unrelated parallel query crashes.

2. If the query engine crashes and existing queries or queries running in parallel crash and there's no fix at the moment, an improvement would be to say something like: `This query crashed due to a problem in a query unrelated to this one. The cause of the unrelated query crashing is ...`. In my case, it became clear because of the context that it was an unrelated query but it could cause confusion if the field also existed in the current query. A user could go down a rabbit hole when the issue itself was in a different query.

## Prisma information

The bug reporting is not related to the schema.

## Environment & setup

- OS: Mac OS
- Database: PostgreSQL
- Prisma version: 2.0.0-beta.8
- Node.js version: v12.16.2

**Comment by janpio** [comment]
Does `prisma.users.findOne` always crash with this message? Or do you have to run the other query that actually crashes with this error message first to make it crash to cause the crash here? Can you run any other queries after the unrelated crash at all?

**Comment by thesunny** [comment]
@janpio 

The **query** that the error gets thrown from is not always from that particular query. The error currently is always `received null value for field last_sent_user which may not be null`. I've documented this in another issue as fixing the errors seems important.

This particular issue is a notice that an error from another query that panics is being reported to a different query.

Perhaps it should be clarified that I am using a single instance of `PrismaClient` in a `Promise.all` (same instance is being used in multiple calls). This seems to be the recommended practice though given the RPC API example provided by Prisma.

**Comment by janpio** [comment]
Thanks.

> 1. Does `prisma.users.findOne` always crash with this message? 
> 2. Or do you have to run the other query that actually crashes with this error message first to make it crash to cause the crash here? 
> 3. Can you run any other queries after the unrelated crash at all?

It would be helpful if you could answer these questions.

**Comment by thesunny** [comment]
@janpio,

1. To clarify, the page always crashes, but it is not always on `prisma.users.findOne`. Sometimes it crashes on a different prisma call but the error message is always from `prisma.book_share_email_invitations.findOne`. I know this because it complains about `last_sent_user` which does not exist in `prisma.users.findOne`. It appears that whenever the call to `prisma.book_share_email_invitations.findOne` crashes, the error message attaches itself to the wrong query.
2. I have to run the `prisma.book_share_email_invitations.findOne` to make it crash. If that doesn't run, I don't get any crashes. The error always pops up (ie. it crashing is consistent) but sometimes the error message ends up in the wrong query (ie. the query that reports the error is inconsistent).
3. Yes, I can run queries after the unrelated crash.

**Comment by janpio** [comment]
> 1. It appears that whenever the call to `prisma.book_share_email_invitations.findOne` crashes, the error message attaches itself to the wrong query.

Does it really attach itself to a wrong query, or do just all subsequent queries _after_ the crash get the same error message? Does the `prisma.book_share_email_invitations.findOne` query get the (for it correct) error message as well?

**Comment by pantharshit00** [comment]
Hm, without a reproduction this will be very hard for me to reproduce. I would say you should try this again once with the latest version once.

**Comment by thesunny** [comment]
I will try with the new version and see if it makes a difference in production.

**Comment by janpio** [comment]
Happy to open this again if we get more information.
